### PR TITLE
KREST-826, KREST-837 Reduce coupling to kafka-clients

### PR DESCRIPTION
**Pull request by dimitarndimitrov**
Reduce coupling to `kafka-clients` by providing our own `waitForCondition` test utilities based on `testWithRetry`.

### CODE REVIEW

File: kafka-rest/src/test/java/io/confluent/kafkarest/TestUtils.java [hunk 1]
```diff
@@ -246,28 +238,127 @@ public static void assertPartitionOffsetsEqual(List<PartitionOffset> a, List<Par
...
+      boolean isExponentialBackoff) {
+    long sleepNanos = initialRetryInterval.toNanos();
+    long start = System.nanoTime();
+    long elapsed;
+    long maxElapsed = timeout != null ? timeout.toNanos() : Long.MAX_VALUE;
     for (int i = 0;; i++) {
       try {
         assertions.run();
         return;
+      } catch (NoRetryException e) {
```
Comment (rigelbm) [inline kafka-rest/src/test/java/io/confluent/kafkarest/TestUtils.java:265]: Do we use this `NoRetryException` anywhere? If we don't, I would get rid of it. One less dependency.

**Review by rigelbm — APPROVED** [review]
LGTM

File: kafka-rest/src/test/java/io/confluent/kafkarest/TestUtils.java (hunk 1 above)
Comment (dimitarndimitrov) [reply kafka-rest/src/test/java/io/confluent/kafkarest/TestUtils.java:265]: We do not, and it's not part of the [Kafka 2.7.0 API](https://kafka.apache.org/27/javadoc/index.html?overview-summary.html), but I was afraid that some components we query might throw it.

Still, what you're suggesting is exactly what I wanted to do, so consider it done.

**Comment by dimitarndimitrov** [comment]
There is one test failure - `ConsumerLagsResourceIntegrationTest.getConsumerLag_returnsConsumerLag`. It seems to be entirely unrelated to the changes here though, as it doesn't use `TestUtils`.

**Pull request merged**